hedera-node/src/main/java/com/hedera/services/store/models/TokenRelationship.java

```diff
@@ -104,13 +105,16 @@ private void adjustBalance(long adjustment) {
 	public void validateAndDissociate(final OptionValidator validator,
 			TokenRelationship treasuryRelationShip) {
 		final var treasury = token.getTreasury();
-		validateFalse(!token.isDeleted() && treasury.getId().equals(account.getId()), ACCOUNT_IS_TREASURY);
-		validateFalse(!token.isDeleted() && isFrozen(), ACCOUNT_FROZEN_FOR_TOKEN);
-
+		validateTrue(!token.isDeleted(), TOKEN_WAS_DELETED, () ->
+				"Given token was deleted");
+		validateTrue( !treasury.getId().equals(account.getId()), ACCOUNT_IS_TREASURY, () ->
+				"Given account is treasury");
+		validateTrue( !isFrozen(), ACCOUNT_FROZEN_FOR_TOKEN, () ->
+				"Account is frozen for given token");
 		if(balance > 0) {
 			final var expiry = Timestamp.newBuilder().setSeconds(token.getExpiry()).build();
 			final var isTokenExpired = !validator.isValidExpiry(expiry);
-			validateFalse(!token.isDeleted() && !isTokenExpired, TRANSACTION_REQUIRES_ZERO_TOKEN_BALANCES);
+			validateFalse(isTokenExpired, TRANSACTION_REQUIRES_ZERO_TOKEN_BALANCES);
```

Comment:
This should be `!isTokenExpired` -> reason: Once a token expires (the expiry comes) the user should be able to disassociate (even if he has balance). If the token is not expired and the user has balance - the node must return `TRANSACTION_REQUIRES_ZERO_TOKEN_BALANCES`